linkml_runtime/loaders/loader_root.py

```diff
@@ -120,36 +130,36 @@ def loads(self, source: str, target_class: type[Union[BaseModel, YAMLRoot]], *,
         """
         return self.load(source, target_class, metadata=metadata)
 
-    def _construct_target_class(self, 
-                                data_as_dict: Union[dict, list[dict]],
-                                target_class: Union[type[YAMLRoot], type[BaseModel]]) -> Optional[Union[BaseModel, YAMLRoot, list[BaseModel], list[YAMLRoot]]]:
+    def _construct_target_class(
+        self, data_as_dict: Union[dict, list[dict]], target_class: Union[type[YAMLRoot], type[BaseModel]]
+    ) -> Optional[Union[BaseModel, YAMLRoot, list[BaseModel], list[YAMLRoot]]]:
         if data_as_dict:
             if isinstance(data_as_dict, list):
-               if issubclass(target_class, YAMLRoot):
-                   return [target_class(**as_dict(x)) for x in data_as_dict]
-               elif issubclass(target_class, BaseModel):
-                   return [target_class.model_validate(as_dict(x)) for x in data_as_dict]
-               else:
-                   raise ValueError(f'Cannot load list of {target_class}')
-            elif isinstance(data_as_dict, dict):
+                if issubclass(target_class, YAMLRoot):
```

Comment:
Some weird spacing issues in this function.

Also made the `if/else` cascade more efficient by removing `else` / `elif` after `return` or `raise` stanzas.